adapters/mqtt-vertx-base/src/main/java/org/eclipse/hono/adapter/mqtt/MqttContext.java

```diff
@@ -258,6 +258,18 @@ public ResourceIdentifier topic() {
         return topic;
     }
 
+    /**
+     * Apply a mapped targetAddress to this mqtt context.
+     * <p>
+     * This overwrites the topic determined from the message.
+     *
+     * @param mappedTargetAddress The mappedTargetAddress.
+     * @throws NullPointerException if mappedTargetAddress is {@code null}.
+     */
+    public void applyMappedTargetAddress(final ResourceIdentifier mappedTargetAddress) {
+        this.topic = Objects.requireNonNull(mappedTargetAddress);
```

Comment:
Based on @calohmn 's comment, I also believe that we should put the mappedTargetAddress into a separate field. Can you please change this accordingly?

Reply:
I updated this and included the changes described by @calohmn